src/methods.jl

```diff
@@ -1,13 +1,21 @@
 export FiniteDifferenceMethod, fdm, backward_fdm, forward_fdm, central_fdm, extrapolate_fdm
 
 """
-    add_tiny(x::Union{AbstractFloat, Integer})
+    estimate_magitude(f, x::T) where T<:AbstractFloat
 
-Add a tiny number, 10^{-40}, to a real floating point number `x`, preserving the type. If
-`x` is an `Integer`, it is promoted to a suitable floating point type.
+Estimate the magnitude of `f` in a neighbourhood of `x`, assuming that the outputs of `f`
+have a "typical" order of magnitude. The result should be interpreted as a very rough
+estimate. This function deals with the case that `f(x) = 0`.
 """
-add_tiny(x::T) where T<:AbstractFloat = x + convert(T, 1e-40)
-add_tiny(x::Integer) = add_tiny(float(x))
+function estimate_magitude(f, x::T) where T<:AbstractFloat
+    M = float(maximum(abs, f(x)))
+    M > 0 && (return M)
+    # Ouch, `f(x) = 0`. But it may not be zero around `x`. We conclude that `x` is likely a
+    # pathological input for `f`. Perturb `x`. Assume that the pertubed value for `x` is
+    # highly unlikely also a pathological value for `f`.
+    Δ = convert(T, 0.1) * max(abs(x), one(x))
+    return float(maximum(abs, f(x + Δ)))
```

Comment:
should we make this recursive?
```suggestion
    return estimate_magitude(f, x + Δ)
```

Reply:
Though even recursive won't solve `sin(0.2*π*x)`

Reply:
That would recurse infinitely for the function `x -> 0.0`. Moreover, a function might just actually be `0` in a neighbour around `x`.